packages/patternfly-4/react-core/src/components/Alert/AlertActionCloseButton.tsx

```diff
@@ -9,25 +10,25 @@ interface AlertActionCloseButtonProps extends ButtonProps {
   onClose?: () => void;
   /** Aria Label for the Close button */
   'aria-label'?: string;
-  /** Variant Label for the Close button */
-  variantLabel?: string;
 }
 
 export const AlertActionCloseButton = ({
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
   className = '',
   onClose = () => undefined as any,
   'aria-label': ariaLabel = '',
-  title,
-  variantLabel = '',
   ...props
 }: AlertActionCloseButtonProps) => (
-  <Button
-    variant={ButtonVariant.plain}
-    onClick={onClose}
-    aria-label={ariaLabel === '' ? `Close ${variantLabel} alert: ${title}` : ariaLabel}
-    {...props}
-  >
-    <TimesIcon />
-  </Button>
+  <AlertContext.Consumer>
+    {({ title, variantLabel }) => (
+      <Button
+        variant={ButtonVariant.plain}
+        onClick={onClose}
+        aria-label={ariaLabel === '' ? `Close ${variantLabel} alert: ${title}` : ariaLabel}
```

Comment:
Right now if variantLabel is not set in `Alert` it will use the default value which is set to 
variantLabel = `${capitalize(variant)} alert:`, 
so it comes out like this: `Close Default alert: alert: Default alert title 1`

I think we should not have any hard-coded words here, for example if this should be in another language etc.. it will cause issues.

I think this would be better as `aria-label={ariaLabel || variantLabel}`

And you could put additional wording into the default for variantLabel in the Alert component instead?

Reply:
The hard coded text is only the default and can be overridden by setting a custom aria-label.

The problem with defaulting to variantLabel for the aria-label if ariaLabel is not set by the user is that the variant label alone doesn't distinguish the close button from others when multiple alerts are present. This is critical, otherwise screen reader users don't have an easy way of knowing which close button they are pressing without further navigating the page for surrounding context clues.

I think any changes to the experience should probably go into a separate PR/work-item, and let's just focus on fixing the bug for this one. Does this make sense?